From 2e8c7d8f03cb36864d31eae2572e6d78112a35a9 Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Wed, 22 Jun 2016 00:01:04 +0800
Subject: [PATCH 1215/1240] a3700: remove palladium support

- remove all palladium relevant code
- remove the defconfig for palladium
- remove palladium hex image from makefile

Change-Id: I832b42c4899aaa6f018a4d170e04cb8421636902
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30602
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/cpu/armv8/armada3700/soc-init.c     | 20 ---------------
 arch/arm/dts/Makefile                        |  1 -
 configs/mvebu_armada3700_palladium_defconfig | 36 --------------------------
 drivers/net/mvneta.c                         | 28 --------------------
 include/configs/armada3700.h                 | 38 ----------------------------
 5 files changed, 123 deletions(-)
 delete mode 100644 configs/mvebu_armada3700_palladium_defconfig

diff --git a/arch/arm/cpu/armv8/armada3700/soc-init.c b/arch/arm/cpu/armv8/armada3700/soc-init.c
index 0b91507..21ae0f9 100644
--- a/arch/arm/cpu/armv8/armada3700/soc-init.c
+++ b/arch/arm/cpu/armv8/armada3700/soc-init.c
@@ -66,21 +66,6 @@ int soc_get_id(void)
 void soc_init(void)
 {
 	/* Do early SOC specific init here */
-
-	/* now there is no timer/MPP driver,
-	  * currently we put all this kind of
-	  * configuration here, and will remove
-	  * this after official driver is ready
-	  */
-#ifdef CONFIG_PALLADIUM
-
-#ifdef CONFIG_I2C_MV
-	/* 0xD0013830[10] = 1'b0 (select GPIO pins to use for I2C_1) */
-	writel((readl(0xd0013830) & ~(1 << 10)), 0xd0013830);
-#endif /* CONFIG_I2C_MV */
-
-#endif /* CONFIG_PALLADIUM */
-
 #ifdef CONFIG_MVEBU_A3700_IO_ADDR_DEC
 	init_io_addr_dec();
 #endif
@@ -141,10 +126,6 @@ void i2c_clk_enable(void)
 
 int dram_init(void)
 {
-#ifdef CONFIG_PALLADIUM
-	/* NO DRAM init sequence in Pallaidum, so set static DRAM size of 256MB */
-	gd->ram_size = 0x20000000;
-#else
 	gd->ram_size = 0;
 
 	/* DDR size has been read from dts DDR node in SPL
@@ -155,7 +136,6 @@ int dram_init(void)
 		error("No DRAM banks detected");
 		return 1;
 	}
-#endif
 
 	return 0;
 }
diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index d1fe065..55fe2f4 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -61,7 +61,6 @@ endif #CONFIG_DEVEL_BOARD
 dtb-$(CONFIG_DEVEL_BOARD) += armada-3700-db.dtb \
 			     armada-3700-db-sgmii1.dtb \
 			     armada-3700-db-sfp2_5g.dtb
-dtb-$(CONFIG_PALLADIUM) += armada-3700-palladium.dtb
 endif #CONFIG_TARGET_ARMADA_3700
 
 ifdef CONFIG_TARGET_ARMADA_8K
diff --git a/configs/mvebu_armada3700_palladium_defconfig b/configs/mvebu_armada3700_palladium_defconfig
deleted file mode 100644
index b9b8d44..0000000
--- a/configs/mvebu_armada3700_palladium_defconfig
+++ /dev/null
@@ -1,36 +0,0 @@
-CONFIG_SPL=y
-CONFIG_SYS_EXTRA_OPTIONS="ARM64"
-+S:CONFIG_ARM=y
-+S:CONFIG_TARGET_ARMADA_3700=y
-+S:CONFIG_PALLADIUM=y
-+S:CONFIG_MVEBU_A3700_IO_ADDR_DEC=y
-+S:CONFIG_MVEBU_A3700_MISC_INIT=y
-+S:CONFIG_MVEBU_A3700_CLOCK=y
-+S:CONFIG_MVEBU_A3700_UART=y
-CONFIG_XENON_MMC=y
-+S:CONFIG_DEVEL_BOARD=y
-+S:# CONFIG_MULTI_DT_FILE is not set
-CONFIG_CMD_BDI=y
-CONFIG_CMD_BOOTD=y
-CONFIG_CMD_RUN=y
-CONFIG_CMD_EDITENV=y
-CONFIG_CMD_SAVEENV=y
-CONFIG_CMD_MEMORY=y
-CONFIG_CMD_NET=y
-CONFIG_CMD_PING=y
-+S:CONFIG_OF_CONTROL=y
-+S:CONFIG_OF_EMBED=y
-+S:CONFIG_DEFAULT_DEVICE_TREE="armada-3700-palladium"
-CONFIG_FDT_SIZE=0x1C00
-CONFIG_MVEBU_ADVK_PCIE=y
-CONFIG_MV_INCLUDE_SATA=y
-CONFIG_SCSI_AHCI_PLAT=y
-+S:CONFIG_MVEBU_MBUS=y
-+S:CONFIG_MVEBU_MBUS_SPL_ONLY=y
-CONFIG_MVNETA=y
-CONFIG_MVEBU_NETA_BYPASS_DEC_WIN=y
-CONFIG_I2C_MV=y
-# CONFIG_I2C_MV_PAD_REG is not set
-CONFIG_USB=y
-CONFIG_USB_XHCI_HCD=y
-+S:# CONFIG_MVEBU_SPL_DDR_SUPPORT is not set
diff --git a/drivers/net/mvneta.c b/drivers/net/mvneta.c
index 94e54c1..82173a1 100644
--- a/drivers/net/mvneta.c
+++ b/drivers/net/mvneta.c
@@ -1522,11 +1522,7 @@ static int mvneta_init_u_boot(struct eth_device *dev, bd_t *bis)
 {
 	struct mvneta_port *pp = dev->priv;
 	unsigned long val;
-#ifdef CONFIG_PALLADIUM
-	unsigned long auto_neg_value;
-#else
 	struct phy_device *phydev;
-#endif /* CONFIG_PALLADIUM */
 
 	mvneta_port_power_up(pp, pp->phy_interface);
 
@@ -1555,14 +1551,6 @@ static int mvneta_init_u_boot(struct eth_device *dev, bd_t *bis)
 		} else {
 			/* Set phy address of the port */
 			mvreg_write(pp, MVNETA_PHY_ADDR, pp->phyaddr);
-#ifdef CONFIG_PALLADIUM
-			/* on Palladium, there is no PHY, need to hardcode Link configuration */
-			pp->init = 1;
-			pp->link = 1;
-			mvneta_probe(dev);
-			mvneta_port_up(pp);
-			mvneta_port_enable(pp);
-#else
 			phydev = phy_connect(pp->bus, pp->phyaddr, dev,
 					     pp->phy_interface);
 
@@ -1583,7 +1571,6 @@ static int mvneta_init_u_boot(struct eth_device *dev, bd_t *bis)
 			 * and address decode configuration.
 			 */
 			pp->init = 1;
-#endif
 		}
 	} else {
 		/* Upon all following calls, this is enough */
@@ -1591,15 +1578,6 @@ static int mvneta_init_u_boot(struct eth_device *dev, bd_t *bis)
 		mvneta_port_enable(pp);
 	}
 
-#ifdef CONFIG_PALLADIUM
-	/* on Palladium, there is no PHY, need to hardcode speed to 1G */
-	auto_neg_value = MVNETA_AUTONEG_CFG_FORCE_LINK_UP | MVNETA_AUTONEG_CFG_BYPASS_AUTO_NEG
-			| MVNETA_AUTONEG_CFG_FORCE_LINK_1G | MVNETA_AUTONEG_CFG_FLOW_CTRL_EN
-			| MVNETA_AUTONEG_CFG_FLOW_CTRL_ADVERTIZE | MVNETA_AUTONEG_CFG_FORCE_FULL_DPLX
-			| MVNETA_AUTONEG_CFG_RESERVED;
-	mvreg_write(pp, MVNETA_GMAC_AUTONEG_CONFIG, auto_neg_value);
-#endif
-
 	return 0;
 }
 
@@ -1744,12 +1722,6 @@ int mvneta_initialize_dev(bd_t *bis, unsigned long base_addr, int devnum, int ph
 	dev->send = mvneta_send;
 	dev->recv = mvneta_recv;
 	dev->write_hwaddr = NULL;
-#ifdef CONFIG_PALLADIUM
-	/* on Palladium, there is no mac address in env, so put a value to skip the validation
-	 * otherwise u-boot would fail at common net driver validation.
-	 */
-	dev->enetaddr[1] = 51;
-#endif
 	/*
 	 * The PHY interface type is configured via the
 	 * board specific CONFIG_SYS_NETA_INTERFACE_TYPE
diff --git a/include/configs/armada3700.h b/include/configs/armada3700.h
index 3d183b6..48d38f5 100644
--- a/include/configs/armada3700.h
+++ b/include/configs/armada3700.h
@@ -48,17 +48,12 @@
 /* Plaform */
 #define CONFIG_MARVELL
 
-/* Clock Definitions */
-#ifdef CONFIG_PALLADIUM
-#define COUNTER_FREQUENCY	(6000)
-#else
 /*
  * The counter on A3700 always fed from reference clock (XTAL)
  * However minimal CPU counter prescaler is 2, so the counter
  * frequency will be divided by 2
  */
 #define COUNTER_FREQUENCY	(12500000)
-#endif
 
 /* DRAM detection stuff */
 #define MV_DRAM_AUTO_SIZE
@@ -99,10 +94,6 @@ extern char __end_of_spl[];
 
 #define CONFIG_MVEBU_SPI_A3700
 
-#ifdef CONFIG_PALLADIUM
-#define CONFIG_SPI_FLASH_SPANSION
-#endif /* CONFIG_PALLADIUM */
-
 #endif /* CONFIG_MVEBU_SPI */
 
 /* The MAX_PAYLOAD_SIZE of A3700 should be aligned with the capability
@@ -125,33 +116,4 @@ extern char __end_of_spl[];
 
 #define CONFIG_SYS_PCI_64BIT	/* enable 64-bit PCI resources */
 
-/*
- * SOC specific overrides
- * Add here all config values that differ
- * from the generic value in mvebu-common.h
- */
-#ifdef CONFIG_PALLADIUM
-
-/* Disable I-CACHE for palladium */
-#define CONFIG_SYS_ICACHE_OFF
-
-#undef CONFIG_BOOTDELAY
-#define CONFIG_BOOTDELAY                -1
-
-#undef CONFIG_BAUDRATE
-#ifdef CONFIG_PALLADIUM
-#define CONFIG_BAUDRATE                 300
-#else
-#define CONFIG_BAUDRATE                 4000
-#endif
-
-#undef CONFIG_BOOTARGS
-#define CONFIG_BOOTARGS			"console=ttyS0,300 earlycon=ar3700_uart,0xd0012000 swiotlb=4096"
-
-#undef  CONFIG_EXTRA_ENV_SETTINGS
-#define CONFIG_EXTRA_ENV_SETTINGS	"kernel_addr=0x107ffc0\0"	\
-					"fdt_addr=0x1000000\0"		\
-					"fdt_high=0xa0000000\0"
-#endif /*CONFIG_PALLADIUM*/
-
 #endif /* __ARMADA_3700_H */
-- 
1.9.1

